tests/typecheck_examples.tests

```diff
@@ -0,0 +1,1007 @@
+filemode : args
+args : --config=tests/typecheck_examples.tests.config
+
+Typecheck example file examples/sessions/smtp-dummy.links
+examples/sessions/smtp-dummy.links
+filemode : args
+args : --config=tests/typecheck_examples.tests.config
+
+Typecheck example file examples/sessions/smtp-end.links
+examples/sessions/smtp-end.links
+filemode : args
+args : --config=tests/typecheck_examples.tests.config
+
+Typecheck example file examples/sessions/smtp-factored-more.links
+examples/sessions/smtp-factored-more.links
+filemode : args
+args : --config=tests/typecheck_examples.tests.config
+
+Typecheck example file examples/sessions/smtp-factored.links
+examples/sessions/smtp-factored.links
+filemode : args
+args : --config=tests/typecheck_examples.tests.config
+
+Typecheck example file examples/sessions/smtp-formlet.links
+examples/sessions/smtp-formlet.links
+filemode : args
+args : --config=tests/typecheck_examples.tests.config
+
+Typecheck example file examples/sessions/smtp.links
+examples/sessions/smtp.links
+filemode : args
+args : --config=tests/typecheck_examples.tests.config
+
+Typecheck example file examples/sessions/state.links
+examples/sessions/state.links
+filemode : args
+args : --config=tests/typecheck_examples.tests.config
+
+Typecheck example file examples/sessions/two-factor-cp.links
+examples/sessions/two-factor-cp.links
+filemode : args
+args : --config=tests/typecheck_examples.tests.config
+
+Typecheck example file examples/sessions/two-factor.links
+examples/sessions/two-factor.links
+filemode : args
+args : --config=tests/typecheck_examples.tests.config
+
+Typecheck example file examples/shredding.links
+examples/shredding.links
+filemode : args
+args : --config=tests/typecheck_examples.tests.config
+ignore : broken example
+
+Typecheck example file examples/shredding2.links
+examples/shredding2.links
+filemode : args
+args : --config=tests/typecheck_examples.tests.config
+ignore : broken example
+
+Typecheck example file examples/shredding3.links
+examples/shredding3.links
+filemode : args
+args : --config=tests/typecheck_examples.tests.config
+ignore : broken example
+
+Typecheck example file examples/shredding4.links
+examples/shredding4.links
+filemode : args
+args : --config=tests/typecheck_examples.tests.config
+ignore : broken example
+
+Typecheck example file examples/silly-progress.links
+examples/silly-progress.links
+filemode : args
+args : --config=tests/typecheck_examples.tests.config
+
+Typecheck example file examples/test-prims.links
+examples/test-prims.links
+filemode : args
+args : --config=tests/typecheck_examples.tests.config
+ignore : broken example
+
+Typecheck example file examples/todo.links
+examples/todo.links
+filemode : args
+args : --config=tests/typecheck_examples.tests.config
+
+Typecheck example file examples/todoDb.links
+examples/todoDb.links
+filemode : args
+args : --config=tests/typecheck_examples.tests.config
+
+Typecheck example file examples/tofix/dom-send.links
```

Comment:
I see little point in adding examples under the directory `tofix` to the test suite. Those examples have been bit rotting for ages now.

Reply:
So far, my approach was to be very explicit and add all examples to the test file. This would ease adding new examples utomatically, because we only need to check for examples whose path is not in the .tests file, yet. However, as @dhil suggested, we may use a more clever script which already does type-checks all new examples and only adds new + working examples.

Reply:
What I suggested was a little different, actually, just a slight variation of your previous script. I think it is fine to conservatively assume an example that fails to type check is "broken". However, you should filter the generated `.tests` file such that it doesn't include directories like `tofix` or experimental stuff like `handlers/` (arguably experimental or incomplete examples should be placed elsewhere)

Reply:
Okay. We removed tofix in the meantime. Are all of the examples in the handlers subdirectory experimental?

Reply:
I did some cleaning recently (as in a couple of months ago). They might just all work now... You can give it a try if you like.

Reply:
The following handler examples are currently broken:
!FAILURE: Typecheck example file examples/handlers/choose.links
!FAILURE: Typecheck example file examples/handlers/count_web.links
!FAILURE: Typecheck example file examples/handlers/deep_pipes.links
!FAILURE: Typecheck example file examples/handlers/delimitedcontinuations.links
!FAILURE: Typecheck example file examples/handlers/nim.links
!FAILURE: Typecheck example file examples/handlers/pipes.links
!FAILURE: Typecheck example file examples/handlers/shallow_pipes.links
!FAILURE: Typecheck example file examples/handlers/tests.links
!FAILURE: Typecheck example file examples/handlers/toggle.links
!FAILURE: Typecheck example file examples/handlers/toggle_web.links


using the following configuration and testing on the current master:
typecheck_only=true
modules=true
enable_handlers=true
shredding=true
session_exceptions=true

If that's fine, I'll just mark those as ignored. The actual output is as follows:
[output.txt](https://github.com/links-lang/links/files/2332711/output.txt)

Reply:
Are those results from the HEAD of master?

Reply:
yes